core/controllers/acl_decorators.py

```diff
@@ -1963,14 +1916,10 @@ def can_user_publish_skill(user, skill_rights):
         Returns:
             bool. Whether the given user can publish the given skill.
         """
-
-        if skill_rights is None:
-            return False
         if role_services.ACTION_PUBLISH_OWNED_SKILL not in user.actions:
             return False
         if skill_rights.is_creator(user.user_id):
             return True
-        return False
```

Comment:
Ditto, if this is unreachable then maybe the logic above needs to be restructured. /cc @aks681

Reply:
Reverted it and added test for the same(my mistake I should not have removed it)